x/axelarcork/types/proposal.go

```diff
@@ -0,0 +1,240 @@
+	return &RemoveAxelarManagedCellarIDsProposal{
+		Title:       title,
+		Description: description,
+		CellarIds:   cellarIds,
+		ChainName:   chainName,
+		ChainId:     chainID,
+	}
+}
+
+func (m *RemoveAxelarManagedCellarIDsProposal) ProposalRoute() string {
+	return RouterKey
+}
+
+func (m *RemoveAxelarManagedCellarIDsProposal) ProposalType() string {
+	return ProposalTypeRemoveManagedCellarIDs
+}
+
+func (m *RemoveAxelarManagedCellarIDsProposal) ValidateBasic() error {
+	if err := govtypes.ValidateAbstract(m); err != nil {
+		return err
+	}
+
+	if len(m.CellarIds.Ids) == 0 {
+		return fmt.Errorf("can't have a remove prosoposal with no cellars")
+	}
+
+	return nil
+}
+
+func NewScheduledCorkProposal(title string, description string, blockHeight uint64, chainName string, chainID uint64, targetContractAddress string, contractCallProtoJSON string) *AxelarScheduledCorkProposal {
+	return &AxelarScheduledCorkProposal{
+		Title:                 title,
+		Description:           description,
+		BlockHeight:           blockHeight,
+		ChainName:             chainName,
+		ChainId:               chainID,
+		TargetContractAddress: targetContractAddress,
+		ContractCallProtoJson: contractCallProtoJSON,
+	}
+}
+
+func (m *AxelarScheduledCorkProposal) ProposalRoute() string {
+	return RouterKey
+}
+
+func (m *AxelarScheduledCorkProposal) ProposalType() string {
+	return ProposalTypeScheduledCork
+}
+
+func (m *AxelarScheduledCorkProposal) ValidateBasic() error {
+	if err := govtypes.ValidateAbstract(m); err != nil {
+		return err
+	}
+
+	if len(m.ContractCallProtoJson) == 0 {
+		return sdkerrors.Wrapf(ErrInvalidJSON, "cannot have empty contract call")
+	}
+
+	if !json.Valid([]byte(m.ContractCallProtoJson)) {
+		return sdkerrors.Wrapf(ErrInvalidJSON, "%s", m.ContractCallProtoJson)
+	}
+
+	if !common.IsHexAddress(m.TargetContractAddress) {
+		return sdkerrors.Wrapf(ErrInvalidEVMAddress, "%s", m.TargetContractAddress)
+	}
+
+	return nil
+}
+
+func NewCommunitySpendProposal(title string, description string, recipient string, chainID uint64, chainName string, amount sdk.Coin) *AxelarCommunityPoolSpendProposal {
+	return &AxelarCommunityPoolSpendProposal{
+		Title:       title,
+		Description: description,
+		Recipient:   recipient,
+		ChainId:     chainID,
+		ChainName:   chainName,
+		Amount:      amount,
+	}
+}
+
+func (m *AxelarCommunityPoolSpendProposal) ProposalRoute() string {
+	return RouterKey
+}
+
+func (m *AxelarCommunityPoolSpendProposal) ProposalType() string {
+	return ProposalTypeCommunitySpend
+}
+
+func (m *AxelarCommunityPoolSpendProposal) ValidateBasic() error {
+	if err := govtypes.ValidateAbstract(m); err != nil {
+		return err
+	}
+
+	if m.Amount.Amount.IsZero() {
+		return ErrValuelessSend
+	}
+
+	if m.Recipient == "" {
+		return sdkerrors.Wrapf(ErrInvalidEVMAddress, "empty recipient")
+	}
+
+	if !common.IsHexAddress(m.Recipient) {
```

Comment:
Is an empty string not also an invalid hex address, or does it render as all zeroes?

Reply:
You are right, removing the extra check